docs/code-comparisons/kedro.rst

```diff
@@ -0,0 +1,248 @@
+   |                                                         |                                                            |
+   +---------------------------------------------------------+------------------------------------------------------------+
+
+The function implementations are exactly the same. Yet, notice that the function names and docstrings were edited slightly. Imperative approaches like ``Kedro`` typically refer to steps as *tasks*  and prefer verbs to describe "the action of the function". Meanwhile, declarative approaches such as ``Hamilton`` describe steps as *assets* and use nouns to refer to "the value returned by the function". This might appear superficial, but it relates to the difference in **Step 2** and **Step 3**.
+
+2. Assemble dataflow
+~~~~~~~~~~~~~~~~~~~~
+
+With ``Kedro``, you need to take your functions from **Step 1** and create ``node`` objects, specifying the node's name, inputs, and outputs. Then, you create a ``pipeline`` from a set of ``nodes`` and ``Kedro`` assembles the nodes into a DAG. Imperative approaches need to specify how tasks (Kedro nodes) relate to each other.
+
+With ``Hamilton``, you pass the module containing all functions from **Step 1** and let Hamilton create the ``nodes`` and the ``dataflow``. This is possible because in declarative approaches like Hamilton, each function defines a transform **and** its dependencies on other functions. Notice how in **Step 1**, ``model_input_table()`` has parameters ``shuttles_preprocessed`` and ``companies_preprocessed``, which refers to other functions in the module. This contains all the required information to build the DAG.
+
+
+.. table::
+   :align: left
+
+   +---------------------------------------------------------+------------------------------------------------------------+
+   | Kedro (imperative)                                      | Hamilton (declarative)                                     |
+   +=========================================================+============================================================+
+   | .. literalinclude:: _kedro_snippets/kedro_assemble.py   | .. literalinclude:: _kedro_snippets/hamilton_assemble.py   | 
+   |                                                         |                                                            |
+   +---------------------------------------------------------+------------------------------------------------------------+
+
+**Benefits of adopting a declarative approach**
+
+- Less errors since you skip manual node creation (i.e., strings **will** lead to typos).
+- Handle complexity since assembling a dataflow remains the same for 10 or 1000 nodes.
+- Maintainability improves since editing your functions (**Step 1**) modifies the structure of your DAG, removing the pipeline definition as a failure point.
+- Readability improves because you can understand how functions relate to each other without jumping between files.
```

Comment:
```suggestion
- Less errors since you skip manual node creation (i.e., strings **will** lead to typos).
- Handle complexity since assembling a dataflow remains the same for 10 or 1000 nodes.
- Maintainability improves since editing your functions (**Step 1**) modifies the structure of your DAG, removing the pipeline definition as a failure point.
- Readability improves because you can understand how functions relate to each other without jumping between files.
- Your code will map very directly to business concepts and terms.
```